pkg/kubectl/proxy_server.go

```diff
@@ -107,7 +107,10 @@ func (f *FilterServer) HandlerFor(delegate http.Handler) *FilterServer {
 }
 
 func (f *FilterServer) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	host, _, _ := net.SplitHostPort(req.Host)
+	host, _, err := net.SplitHostPort(req.Host)
```

Comment:
I hate to be "that guy" but this could do with being extracted to a function and unit tested.  Is this the only failure mode that SplitHostPort can experience?

I'd accept a TODO and a promise to followup...